### PR TITLE
Add stack transaction id array to context to improve stack trace

### PR DESCRIPTION
**Pull request by leolassence**

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -94,12 +94,36 @@ const emptyTransport = {
     warn: noop
 };
 
+function generateStackId() {
+    let id = '';
+    const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789';
```
Comment (PuKoren) [inline src/utils.js]: You can put `chars` and `chars.length` into external variables so we do not instanciate a new array on each call

File: src/utils.js [hunk 2]
```diff
@@ -94,12 +94,36 @@ const emptyTransport = {
...
+        id += chars.charAt(Math.floor(Math.random() * chars.length));
+    }
+
+    return id;
+}
+
+function stackTransactionId(context) {
+    let stack;
+
+    if (!context.transactionStack || context.transactionStack.length === 0) {
```
Comment (PuKoren) [inline src/utils.js]: I think the second part of the `if` is not usefull here as `[...[], 'abcd']` will produce `['abcd']`
Unless you want to make sure it is an array, if so better use `typeof context.transactionStack === 'array'`

File: src/index.js [hunk 3]
```diff
@@ -13,7 +13,8 @@ const {
     deserializeError,
     serializeError,
     extend,
-    emptyTransport
+    emptyTransport,
+    stackTransactionId
```
Comment (PuKoren) [inline src/index.js]: Maybe we should find a more explicit name for this function, like `getTransactionStack` ? @nbeydoon

File: src/utils.js (hunk 1 above)
Comment (leolassence) [reply src/utils.js]: ok

File: src/utils.js (hunk 2 above)
Comment (leolassence) [reply src/utils.js]: ok

File: src/index.js (hunk 3 above)
Comment (leolassence) [reply src/index.js]: ok

**Pull request merged**